src/collective/ifttt/browser/ifttt_trigger.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+
+from collective.ifttt import _
+from plone import api
+from plone.autoform.form import AutoExtensibleForm
+from z3c.form import button
+from z3c.form import form
+from zope import schema
+from zope.globalrequest import getRequest
+from zope.interface import Interface
+
+
+class AddRuleSchema(Interface):
+    '''
+        Define schema for add rule form
+        '''
+
+    ifttt_event_name = schema.TextLine(
+        title=_(u'IFTTT applet name'),
+        description=_(
+            u'Give the name of IFTTT applet which you want to trigger'
+        ),
+        required=True,
+    )
+
+    content_types = schema.Tuple(
+        title=_(u'Restrict Content Types'),
+        description=_(
+            u'Select certain content types which should be restricted'
+        ),
+        required=False,
+        missing_value=None,
+        default=(),
+        value_type=schema.Choice(
+            vocabulary='plone.app.vocabularies.ReallyUserFriendlyTypes'
+        )
+    )
+
+    workflow_transitions = schema.Tuple(
+        title=_(u'Restrict Workflow Transitions'),
+        description=_(
+            u'Select certain workflow transitions which should be restricted'
+        ),
+        required=False,
+        missing_value=None,
+        default=(),
+        value_type=schema.Choice(
+            vocabulary='plone.app.vocabularies.WorkflowTransitions'
+        )
+    )
+
+
+class AddRule(AutoExtensibleForm, form.Form):
+    '''
+    Define Form
+    '''
+
+    schema = AddRuleSchema
+    ignoreContext = True
+    form_name = 'add_ifttt_rule'
+
+    label = _(u'Add new IFTTT trigger')
+    description = _(u'This will add new IFTTT Trigger')
+
+    def update(self):
+        # disable Plone's editable border
+        self.request.set('disable_border', True)
+
+        # call the base class version - this is very important!
+        super(AddRule, self).update()
+
+    @button.buttonAndHandler(_(u'ADD'))
+    def handleApply(self, action):
+        data, errors = self.extractData()
+        if errors:
+            self.status = self.formErrorsMessage
+            return
+
+        try:
+            # all the backend magic goes here
+
+            # Redirect back to the front page with a status message
+
+            api.portal.show_message(
+                message=_(
+                    u'Successfully applied the IFTTT applet '
+                    u'${ifttt_event_name} to ${title}',
+                    mapping=dict(
+                        ifttt_event_name=data['ifttt_event_name'],
+                        title=self.context.Title().decode('utf-8', 'ignore'),
+                    ),
+                ),
+                request=getRequest(),
+                type='info'
+            )
+
+        except TypeError:
```

Comment:
Here you may be guessing possible exceptions too early. Actually the view code at `python:path('object/@@checkifttt').check_iftttconfig()` should take care that this code is only called when it can execute successfully (by creating, configuring and assigning a content rule for the content folder - IFTTT is only called later - outside this code).

Reply:
So for the required checks. Should I run a test for creating, configuring and assigning a content rule to the given folder under check_iftttconfig() function?

Reply:
Not really. But right now this looks for me like you are guessing exception that might rise. If you don't know for sure, which excaptions might rise, but still want to handle them, try:

```python
try:
     # code
except Exception:
    logger.exception('Unexpected exception:')
    # notification code
```

So, 1) you may capture all exceptions (you may need to add `  # noqa: CODE` at the end of that except line if code analysis tests complain) 2) but remember to log the complete exception so that those won't go unnoticed.